docs/topics/unused-return-value-checker.md

```diff
@@ -0,0 +1,198 @@
+<tab id="maven" title="Maven">
+
+```xml
+<!-- pom.xml -->
+<plugin>
+    <groupId>org.jetbrains.kotlin</groupId>
+    ..
+    <configuration>
+        <args>
+            <arg>-Xreturn-value-checker=check</arg>
+        </args>
+    </configuration>
+</plugin>
+```
+
+</tab>
+</tabs>
+
+## Mark functions to check ignored results
+
+When you set [the `-Xreturn-value-checker` compiler option](#configure-the-unused-return-value-checker) to `check`,
+the checker reports ignored results only from expressions that are marked, like most functions in the Kotlin standard library.
+
+To mark your own code, 
+use the [`@MustUseReturnValues`](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin/-must-use-return-value/) annotation.
+You can apply it to a file, class, or function depending on the scope you want the checker to cover.
+
+For example, you can mark an entire file:
+
+```kotlin
+// Marks all functions and classes in this file so the checker reports unused return values
+@file:MustUseReturnValues
+
+package my.project
+
+fun someFunction(): String
+```
+
+Or a specific class:
+
+```kotlin
+// Marks all functions in this class so the checker reports unused return values
+@MustUseReturnValues
+class Greeter {
+    fun greet(name: String): String = "Hello, $name"
+}
+
+fun someFunction(): Int = ...
+```
+
+> You can apply the checker to your entire project by setting the `-Xreturn-value-checker` compiler option to `full`.
+> With this option, you don't have to annotate your code with `@MustUseReturnValues`.
+>
+{style="note"}
+
+## Suppress reports for ignored results
+
+You can suppress reports on specific functions by annotating them with [`@IgnorableReturnValue`](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin/-ignorable-return-value/).
+Annotate functions where ignoring the result is common and expected, such as `MutableList.add`:
+
+```kotlin
+@IgnorableReturnValue
+fun <T> MutableList<T>.addAndIgnoreResult(element: T): Boolean {
+    return add(element)
+}
+```
+
+You can suppress a warning without annotating the function itself.
+To do this, assign the result to a special unnamed variable with an underscore syntax (`_`):
+
+```kotlin
+// Non-ignorable function
+fun computeValue(): Int = 42
+
+fun main() {
+
+    // Reports a warning: result is ignored
+    computeValue()
+
+    // Suppresses the warning only at this call site with a special unused variable
+    val _ = computeValue()
+}
+```
+
+### Ignored results in function overrides
+
+When you override a function, the override inherits the reporting rules defined by the annotations on the base declaration.
+This also applies when the base declaration is part of the Kotlin standard library or of other library dependencies, so the checker reports ignored results for overrides of functions like `Any.hashCode()`.
+
+Additionally, you can't override a function marked with `@IgnorableReturnValue` with one that [requires its return value to be used](#mark-functions-to-check-ignored-results).
```

Comment:
by "one" do we refer to "another function". What about explicitly mention it to be clearer?